### PR TITLE
Fix guide on how to integrate `pallet-contracts`

### PR DESCRIPTION
**Pull request by cmichi**

### CODE REVIEW

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx [hunk 1]
```diff
@@ -56,24 +87,21 @@ Implement the Contracts pallet [configuration traits][contracts-config-rustdocs]
 
    ```rust
    /* --snip-- */
+   Contracts: pallet_contracts,
```
Comment (NukeManDan) [inline v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx:90]: This was just patched based on #1042 , perhaps I missed why this is relevant to parameter types

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx [hunk 2]
```diff
@@ -88,110 +116,196 @@ Implement the Contracts pallet [configuration traits][contracts-config-rustdocs]
...
+The Contracts pallet exposes custom runtime APIs and RPC endpoints.
+
+An example of such an API + RPC is to execute a contract call as a dry-run ‒ so without
+actually modifying any storage.
+This dry-run is often used by off-chain tooling (e.g. a frontend) to display a contracts state.
+Think of an ERC-20 contract, in order to display the balance of an account the frontend
+would execute a dry-run of the balance function via this RPC.
+
+In order to use the Contracts pallet for this you'll use two other pallets: the
+Contracts RPC Runtime API pallet and the Contracts RPC pallet.
```
Comment (NukeManDan) [inline v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx:154]: One sentence per line throughout please, we don't use hard wrap on character count.

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx [hunk 3]
```diff
@@ -88,110 +116,196 @@ Implement the Contracts pallet [configuration traits][contracts-config-rustdocs]
...
+
 
 [create-first-chain-tutorial]: /tutorials/v3/create-your-first-substrate-chain
-[contracts-config-rustdocs]: /rustdocs/latest/pallet_contracts/trait.Config.html
-[contracts-frame-gh]: https://github.com/paritytech/substrate/blob/master/frame/contracts/src/lib.rs#L144
-[bin-runtime-contracts-frame]: https://github.com/paritytech/substrate/blob/632b32300eb9376767c2ae7b38e79b3f7f5329b1/bin/node/runtime/src/lib.rs#L887-L903
-[contracts-api-rustdocs]: /rustdocs/latest/pallet_contracts_rpc_runtime_api/trait.ContractsApi.html
+[contracts-config-rustdocs]: /docs.substrate.io/rustdocs/latest/pallet_contracts/pallet/trait.Config.html
+[contracts-frame-gh]: https://github.com/paritytech/substrate/blob/master/frame/contracts/src/lib.rs
+[bin-runtime-contracts-frame]: https://github.com/paritytech/substrate/blob/b75a253f148aa36fa17cf795b9f2fc2f22d0fcc5/bin/node/runtime/src/lib.rs#L1078-L1095
```
Comment (NukeManDan) [inline v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx:301]: Should all links be to a specific tag? Ideally based on a concrete release of the tooling and ink. So we can reason about things easier and maintain on releases

**Review by NukeManDan — CHANGES_REQUESTED** [review]

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx [hunk 4]
```diff
@@ -10,34 +10,58 @@ section: how to guides
...
-This guide will show you how to add the [Contracts pallet](/rustdocs/latest/pallet_contracts/index.html) to your runtime so that your blockchain can support Wasm smart contracts.
+This guide will show you how to add the [Contracts pallet](/rustdocs/latest/pallet_contracts/index.html) to your runtime so that your blockchain can support WebAssembly (Wasm) smart contracts.
+
 You can follow similar patterns to add additional FRAME pallets to your runtime, however you should note that each pallet is a little different in terms of the specific configuration settings needed to use it correctly.
 
+## What is the Contracts Pallet for?
+
+Broadly speaking there are two big use-cases you can apply it to.
+We'll only shortly describe those, for more information please check
+[the pallet reamde][pallet-readme].
```
Comment (athei) [inline v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx:21]:
```suggestion
[the pallet readme][pallet-readme].
```

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx [hunk 5]
```diff
@@ -10,34 +10,58 @@ section: how to guides
...
+### Smart contracts as first-class citizens
+
+You want to build a general-purpose smart contracts chain, with them being the main focus of your chain.
+Maybe you have an innovative idea and want to have a smart contracts base layer which you complement
+with your chains unique selling point.
+
+### Smart contracts for users to customize how they interact with your chain
+
+You've built a chain that doesn't have smart contracts as first-class citizens,
+instead most of the logic is built using Substrate pallets that use weight-metering.
```
Comment (athei) [inline v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx:32]: Weight-metering isn't a thing. Metering is understood by most people as "high overhead" technique what pallets are avoiding.

```suggestion
instead most of the logic is built using Substrate pallets.
```

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx (hunk 1 above)
Comment (athei) [reply v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx:90]: Yeah this snippet doesn't look like it belongs here.

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx [hunk 6]
```diff
@@ -10,34 +10,58 @@ section: how to guides
...
 ## Import the dependencies
 
-1. Add Contracts to your runtime.
-   To learn how to include Contracts in your runtime, see [Basic pallet integration](/how-to-guides/v3/basics/pallet-integration).
-
-1. Update `runtime/Cargo.toml` with the following entries:
+1. Add the following dependencies to your `runtime/Cargo.toml`:
    - `pallet-contracts`
    - `pallet-contracts-primitives`
+1. Integrate the Contracts pallet into your runtime as described in the
```
Comment (athei) [inline v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx:54]: The two RPC crates are missing here.

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx [hunk 7]
```diff
@@ -10,34 +10,58 @@ section: how to guides
...
 
 ## Import the dependencies
 
-1. Add Contracts to your runtime.
-   To learn how to include Contracts in your runtime, see [Basic pallet integration](/how-to-guides/v3/basics/pallet-integration).
-
-1. Update `runtime/Cargo.toml` with the following entries:
+1. Add the following dependencies to your `runtime/Cargo.toml`:
    - `pallet-contracts`
    - `pallet-contracts-primitives`
```
Comment (lisa-parity) [inline v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx:53]:
```suggestion
   - `pallet-contracts-primitives`
   - `pallet-contracts-rpc-runtime-api`
   - `pallet-contracts-rpc`
```

**Review by lisa-parity — APPROVED** [review]

File: v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx (hunk 6 above)
Comment (cmichi) [reply v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx]: They are added in a special section below (not my idea, the original how-to had this structure).

**Review by athei — APPROVED** [review]

**Pull request merged**